### PR TITLE
Fix expression incompatible between parser and executor

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #26275 
/kind bug

### CODE REVIEW

**Comment by longjiquan** [comment]
rerun ut

File: internal/parser/planparserv2/plan_parser_v2.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func ParseExpr(schema *typeutil.SchemaHelper, exprStr string) (*planpb.Expr, err
 	if predicate == nil {
 		return nil, fmt.Errorf("cannot parse expression: %s", exprStr)
 	}
-	if !typeutil.IsBoolType(predicate.dataType) {
+	if !canBeExecuted(predicate) {
```
[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/parser/planparserv2/plan_parser_v2.go:65]: Yes, you're right for the column expression. But only judging column expression is not enough, value expression and binary arithmetic expression themselves can't be a node neither.

[the remark this answers is not in the document]
Comment (xiaocai2333) [reply internal/parser/planparserv2/plan_parser_v2.go:65]: ok

**Pull request merged**